axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java

```diff
@@ -457,24 +456,19 @@ public LocalDateTime computePlannedEndDateT(OperationOrder operationOrder)
 
   public long computeEntireCycleDuration(OperationOrder operationOrder, BigDecimal qty)
       throws AxelorException {
-
-    long machineDuration = this.computeMachineDuration(operationOrder, qty);
-
-    long humanDuration = this.computeHumanDuration(operationOrder, qty);
-
-    if (machineDuration >= humanDuration) {
-      return machineDuration;
-    } else {
-      return humanDuration;
-    }
-  }
-
-  public long computeMachineDuration(OperationOrder operationOrder, BigDecimal qty)
-      throws AxelorException {
     ProdProcessLine prodProcessLine = operationOrder.getProdProcessLine();
     WorkCenter workCenter = prodProcessLine.getWorkCenter();
 
     long duration = 0;
+    
+    BigDecimal maxCapacityPerCycle = prodProcessLine.getMaxCapacityPerCycle();
+
+    BigDecimal nbCycles;
+    if (maxCapacityPerCycle.compareTo(BigDecimal.ZERO) == 0) {
+  	  nbCycles = new BigDecimal(1);
```

Comment:
BigDecimal.ONE

Reply:
It was in the original code. I guessed that if you put 0 as the maxCapacity it should mean that you can do one product at a time so the number of cycles is the qty (then I think that there is a mistake nbCycles should be equal to qty and not to BigDecimal(1)).

Reply:
No, I means that newBigDecimal(1) can be replaced per BigDecimal.ONE

Reply:
Ok got it so it is not necessary now